Plutarch/DataRepr/Internal.hs

```diff
@@ -276,8 +275,7 @@ pmatchDataSum d handlers =
 -}
 newtype PIsDataReprInstances (a :: PType) (s :: S) = PIsDataReprInstances (a s)
 
--- Q: Why are these superclass constraints here?
-class (PMatch a, PIsData a) => PIsDataRepr (a :: PType) where
+class (PlutusType a, PIsData a) => PIsDataRepr (a :: PType) where
```

Comment:
I just realised this is a breaking change... sorry. Can we revert this commit? lol

Reply:
Hmm, breaking because users may not be implementing `PlutusType` yet? Don't we *want* them to do just that though, from now on?

But yes, we can revert it if you want.

Reply:
Existing code would fail to compile because `PlutusType` isn't derived.

Reply:
Reverted - made a note to change this in the future.